### DISCUSSION TITLE
Consider the range 100.64.x.x private

### DISCUSSION THREAD

**Discussion by kanor1306** [keycloak/keycloak#15400] [Ideas]
According to the documentation in [SSL Modes](https://www.keycloak.org/docs/latest/server_admin/#_ssl_modes):

> - External requests: Users can interact with Keycloak without SSL so long as they stick to private IP addresses such as localhost, 127.0.0.1, 10.x.x.x, 192.168.x.x, and 172.16.x.x. If you try to access Keycloak without SSL from a non-private IP address, you will get an error.

For reasons related with the design of my system sometimes I have Kubernetes clusters in which  I don't want to use SSL for internal communication and where pods get IPs in the range `100.64.x.x`. In these cases, I would like internal services to access Keycloak without using SSL, and currently this is not possible as their requests are considered external (and setting the SSL mode to `None` is not an option). 

I know that 100.64.x.x is not exactly a private range (as per IANA definitions), but if I am not mistaken it is not Internet routable either, so there is no chance for this range to be public IPs (though I may be wrong in this point)

Alternatively it would be nice to be able to configure a IP range in the SSL modes, but I would like to hear feedback about these two things before attempting a PR

Cheers!

**Comment by korenlev** [comment]
https://github.com/keycloak/keycloak/issues/25881 as reference , and a fix here: https://github.com/keycloak/keycloak/pull/26316  , please support it

**Reply by kanor1306** [reply]
Thanks for the heads up @korenlev

**Comment by stianst** [comment]
We're not going to add support for this as TLS should really be used in anything beyond local development.
